Tests/UnitTests/Core/AdyenContextTests.swift

```diff
@@ -54,6 +54,31 @@ class AdyenContextTests: XCTestCase {
         let context = AdyenContext(apiContext: apiContext, payment: Dummy.payment)
         XCTAssertNil(context.analyticsProvider)
     }
+    
+    func testBothAnalyticsProviderShouldBeCreated() {
```

Comment:
This "both" triggers a set of questions to anyone seeing this for the first time. Why do we have two analytic providers? How do I know which one to use?

Do you think we can improve this even more by extracting initial analytics call to a separate entity? Then even the variable name would give a lot more context (like `initialDataAnalyticsProvider`)

Reply:
I see the problem, right!

I would play around with completely extracting that single call into a separate entity and calling it once. It would allow to declutter **main** analytics provider (that could be something like `AnalyticsCoordinator`, this is a long shot, I just try to show my thinking). 

By doing so you can hide `UniqueAssetAPIClient` and `InitialAnalyticsRequest` inside that entity hence declutter main analytics provider.
Apart from that I guess we could try to get rid of `func sendInitialAnalytics()` interface method and just send this analytics upon the new entity instantiation, I assume we have all necessary data to create the initial request and we don't actually need to wait for any additional input or network response.